traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go

```diff
@@ -149,14 +135,18 @@ func Update(w http.ResponseWriter, r *http.Request) {
 	}
 
 	var origSC TOServiceCategory
-	if err := inf.Tx.QueryRow(`SELECT name FROM service_category WHERE name = $1`, name).Scan(&origSC.Name); err != nil {
+	if err := inf.Tx.QueryRow(`SELECT name, last_updated FROM service_category WHERE name = $1`, name).Scan(&origSC.Name, &origSC.LastUpdated); err != nil {
 		if err == sql.ErrNoRows {
 			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("no service category found with name "+name), nil)
 			return
 		}
 		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
 		return
 	}
+	if !api.IsUnmodified(r.Header, origSC.LastUpdated.Time) {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusPreconditionFailed, errors.New("service category has been modified"), nil)
```

Comment:
The API error should not be "service category has been modified", rather it should be something like "service category could not be modified because the precondition failed", or something to that effect.

Reply:
Fixed in https://github.com/apache/trafficcontrol/pull/5231/commits/7230f480a8a75cdf9c14c8ceb2f95f88ee02b26d